plugins/Utils/windowstatestorage.cpp

```diff
@@ -16,167 +17,319 @@
 
 #include "windowstatestorage.h"
 
-#include <QtConcurrent>
 #include <QDebug>
+#include <QDir>
+#include <QMetaObject>
+#include <QObject>
 #include <QSqlQuery>
 #include <QSqlError>
 #include <QSqlResult>
+#include <QStandardPaths>
 #include <QRect>
 #include <unity/shell/application/ApplicationInfoInterface.h>
 
-QMutex WindowStateStorage::s_mutex;
+AsyncQuery::AsyncQuery(const QString& dbName):
+    m_dbName(dbName)
+{
+}
 
-inline QString sanitiseString(QString string) {
-    return string.remove(QLatin1Char('\"'))
-                 .remove(QLatin1Char('\''))
-                 .remove(QLatin1Char('\\'));
+AsyncQuery::~AsyncQuery()
```

Comment:
Should AsyncQuery be in its own \_p header file? How do I do that correctly?

Reply:
Imo we should just place it in on top of the cpp file, it is not shared anywhere/

Reply:
I have moved AsyncQuery out of the header file. It is now declared in the .cpp.